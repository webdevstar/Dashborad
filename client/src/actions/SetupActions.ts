import alt, { AbstractActions } from '../alt';
import * as request from 'xhr-request';
import { IToast, ToastActions } from '../components/Toast';

interface ISetupActions {
  load(): any;
  save(setupConfig: ISetupConfig, successCallback: () => void): any;
  failure(error: any): void;
}

class SetupActions extends AbstractActions implements ISetupActions {

  load() {
    
    return (dispatcher: (setupConfig: ISetupConfig) => void) => {
      
      request('/api/setup', { json: true }, (setupError: any, setupConfig: ISetupConfig) => {
          return dispatcher(setupConfig);
        });
    };
  }

  save(setupConfig: ISetupConfig, successCallback: () => void) {
    return (dispatcher: (setupConfig: ISetupConfig) => void) => {

      setupConfig.stage = 'configured';
      let stringConfig = JSON.stringify(setupConfig);
      
      request('/api/setup', {
          method: 'POST',
          json: true,
          body: { json: stringConfig }
        }, 
              (setupError: any, setupJson: any) => {

          if (setupError) {
            return this.failure(setupError);
          }

          return request('/auth/init', 
                         (authError: any, authJson: any) => {

              if (authError) {
                return this.failure(authError);
              }

              let toast: IToast = { text: 'Setup was saved successfully.' };
              ToastActions.addToast(toast);

              try {
                if (successCallback) {
                  successCallback();
                }
              } catch (e) { }

              return dispatcher(authJson);
            }
          );
        }
      );
    };    
  }

  failure(error: any) {
    return { error };
  }
}

const setupActions = alt.createActions<ISetupActions>(SetupActions);

export default setupActions;
