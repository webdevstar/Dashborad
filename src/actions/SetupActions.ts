import alt, { AbstractActions } from '../alt';
import * as request from 'xhr-request';
import {IToast, ToastActions} from '../components/Toast';

interface ISetupActions {
  load(): any;
  save(setupConfig: ISetupConfig, successCallback: () => void): any;
  failure(error: any): void;
}

class SetupActions extends AbstractActions implements ISetupActions {
  constructor(alt: AltJS.Alt) {
    super(alt);
  }

  load() {
    
    return (dispatcher: (setupConfig: ISetupConfig) => void) => {
      
      request('/api/setup', { json: true },
        (error: any, setupConfig: ISetupConfig) => {
          
          return dispatcher(setupConfig);
        });
    };
  }

  save(setupConfig: ISetupConfig, successCallback: () => void) {
    return (dispatcher: (setupConfig: ISetupConfig) => void) => {

      setupConfig.stage = 'configured';
      let stringConfig = JSON.stringify(setupConfig);
      
      request('/api/setup', {
          method: 'POST',
          json: true,
          body: { json: stringConfig }
        }, 
        (error: any, json: any) => {

          if (error) {
            return this.failure(error);
          }

          return request('/auth/init', 
            (error: any, json: any) => {

              if (error) {
                return this.failure(error);
              }

              let toast : IToast = { text: 'Setup was saved successfully.' };
              ToastActions.addToast(toast);

              try {
                if (successCallback) {
                  successCallback();
                }
              } catch (e) { }

              return dispatcher(json);
            }
          );
        }
      );
    };    
  }

  failure(error: any) {
    return { error };
  }
}

const setupActions = alt.createActions<ISetupActions>(SetupActions);

export default setupActions;
