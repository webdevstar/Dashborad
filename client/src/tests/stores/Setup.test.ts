import SetupStore from "../../stores/SetupStore";
import SetupActions from "../../actions/SetupActions";

describe('Data Source: Samples', () => {

  this.timeout = 10000;
  beforeAll(() => {
  })

  it('Testing SetupActions', (done) => {

    let checkState = (state) => {
      expect(state).toHaveProperty('admins');
      expect(state).toHaveProperty('allowHttp');
      expect(state).toHaveProperty('clientID');
      expect(state).toHaveProperty('clientSecret');
      expect(state).toHaveProperty('enableAuthentication');
      expect(state).toHaveProperty('issuer');
      expect(state).toHaveProperty('loaded');
      expect(state).toHaveProperty('redirectUrl');
      expect(state).toHaveProperty('saveSuccess');
      expect(state).toHaveProperty('stage');

      SetupStore.unlisten(checkState);
      done();
    };

    SetupStore.listen(checkState);
    (SetupActions.load as any).defer();
  });
})
