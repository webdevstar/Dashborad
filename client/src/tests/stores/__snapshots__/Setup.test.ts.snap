// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SetupActionsLoad 1`] = `
Store {
  "admins": Array [],
  "allowHttp": false,
  "clientID": "",
  "clientSecret": "",
  "enableAuthentication": false,
  "issuer": "",
  "loaded": true,
  "redirectUrl": "",
  "saveSuccess": true,
  "stage": "",
}
`;
